src/test/java/net/datafaker/providers/base/PhoneNumberTest.java

```diff
@@ -57,103 +62,139 @@ void testPhone_CA() {
+        for (int i = 0; i < COUNT; i++) {
             String phoneNumber = phoneNumberProvider.phoneNumber();
             try {
                 Phonenumber.PhoneNumber proto = util.parse(phoneNumber, phoneNumberRegion);
                 if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                    if (log.isLoggable(FINE)) {
+                        log.fine("Invalid phone: %s".formatted(phoneNumber));
+                    }
                     errorCount++;
                 }
-            } catch (Exception e) {
+            } catch (NumberParseException e) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s (caused by: %s)".formatted(phoneNumber, e));
+                }
                 errorCount++;
             }
         }
-        assertThat(errorCount).isLessThanOrEqualTo(15);
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
     }
 
     @ParameterizedTest
     @MethodSource("generateLanguageAndRegionOfLocales")
-    void testAllPhoneNumberInternational(Locale locale, String phoneNumberRegion) throws NumberParseException {
-        final PhoneNumberUtil util = PhoneNumberUtil.getInstance();
+    void testAllPhoneNumberInternational(Locale locale, String phoneNumberRegion, int allowedErrorsCount) throws NumberParseException {
         int errorCount = 0;
         final BaseFaker faker = new BaseFaker(locale);
         final PhoneNumber phoneNumberProvider = faker.phoneNumber();
-        for (int i = 0; i < 10; i++) {
+        for (int i = 0; i < COUNT; i++) {
             String phoneNumber = phoneNumberProvider.phoneNumberInternational();
             Phonenumber.PhoneNumber proto = util.parse(phoneNumber, phoneNumberRegion);
             if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s".formatted(phoneNumber));
+                }
                 errorCount++;
             }
         }
-        assertThat(errorCount).isLessThan(25);
+        assertThat(errorCount).isLessThanOrEqualTo(allowedErrorsCount);
+    }
+
+    @ParameterizedTest
+    @MethodSource("generateLanguageAndRegionOfLocales")
+    void testAllPhoneNumberMobile(Locale locale, String phoneNumberRegion, int allowedErrorsCount) throws NumberParseException {
+        int errorCount = 0;
+        final BaseFaker faker = new BaseFaker(locale);
+        final PhoneNumber phoneNumberProvider = faker.phoneNumber();
+        for (int i = 0; i < COUNT; i++) {
+            String phoneNumber = phoneNumberProvider.cellPhone();
+            Phonenumber.PhoneNumber proto = util.parse(phoneNumber, phoneNumberRegion);
+            if (!util.isValidNumberForRegion(proto, phoneNumberRegion)) {
+                if (log.isLoggable(FINE)) {
+                    log.fine("Invalid phone: %s".formatted(phoneNumber));
+                }
+                errorCount++;
```

Comment:
```suggestion
                errorCount++;
                if (allowedErrorsCount > errorCount) {
                    break;
                }    
```
I guess there is no need to iterate if we already exceed the limit

Reply:
Yep, I've added "break" at these places.
Though, this code is temporary anyway. We need to configure phone numbers to achieve 0 error tolerance. :)

Reply:
I think I tried that some time ago, to get close to zero tolerance, but while I managed for some countries, it was non trivial to get them all in shape. Happy if you give it a shot though